api/job_resources.go

```diff
@@ -0,0 +1,28 @@
+package api
+
+import (
+	"github.com/hashicorp/nomad/nomad/structs"
+)
+
+type JobResources struct {
+	client *Client
+}
+
+// JobResources returns a handle on the JobResources endpoints
+func (c *Client) JobResources() *JobResources {
+	return &JobResources{client: c}
+}
+
+// List returns a list of all resources for a particular context. If a
+// context is not specified, matches for all contezts are returned.
+func (j *JobResources) List(prefix, context string) (*structs.ResourceListResponse, error) {
+	var resp structs.ResourceListResponse
+	req := &structs.ResourceListRequest{Prefix: prefix, Context: context}
+
+	_, err := j.client.write("/v1/resources/", req, &resp, nil)
```

Comment:
We renamed it to /v1/cluster/search in a later commit, but this is a new endpoint.